### PR TITLE
Fix part of #295: Change the CircleCI workflow

### PR DESCRIPTION
**Pull request by vinitamurthi**
## Explanation
This PR makes some changes to the circleCI workflow: 
1. Breakdown the tests into separate steps, and add a setup step similar to the web circleCI workflow
2. Create two workflows: One for flaky tests and one for non-flaky tests, non-flaky tests should always pass.
Follow up PRs will slowly move the flaky tests into the non-flaky queue and then we can finally block submit on this
Also fix a network interceptor test
Fixes part of #295

## Checklist

- [ ] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [ ] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [ ] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [ ] The PR is made from a branch that's **not** called "develop".
- [ ] The PR is made from a branch that is up-to-date with "develop".
- [ ] The PR's branch is based on "develop" and not on any other branch.
- [ ] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by vinitamurthi** [comment]
@seanlip @BenHenning PTAL!

File: data/src/test/java/org/oppia/data/backends/test/MockSubtopicTest.kt [hunk 1]
```diff
@@ -40,15 +40,15 @@ class MockSubtopicTest {
...
-    val delegate = mockRetrofit.create(SubtopicService::class.java)
-    val mockSubtopicService = MockSubtopicService(delegate)
-
-    val subtopic = mockSubtopicService.getSubtopic("Subtopic 1", "randomId")
-    val subtopicResponse = subtopic.execute()
-
-    assertThat(subtopicResponse.isSuccessful).isTrue()
-    assertThat(subtopicResponse.body()!!.subtopicTitle).isEqualTo("Subtopic 1")
-  }
+//  @Test
```
Comment (seanlip) [inline data/src/test/java/org/oppia/data/backends/test/MockSubtopicTest.kt]: Weird to see all this commented out. Is it intentional?

Comment (vinitamurthi) [reply data/src/test/java/org/oppia/data/backends/test/MockSubtopicTest.kt]: Oops, that was not intentional..removed it

**Review by seanlip — APPROVED** [review]
LGTM, though with the caveat that I'm not super familiar with CircleCI config -- I'll ping @kevinlee12 to ask if he has any comments based on his experience with oppia-web.

File: data/src/test/java/org/oppia/data/backends/ApiUtils.kt [hunk 2]
```diff
@@ -11,7 +12,11 @@ class ApiUtils {
       val assetsPath: String = "../data/src/test/assets/api_mocks/" + jsonPath
       val inputStream = FileInputStream(assetsPath)
       val json: String = inputStream.bufferedReader().use { it.readText() }
-      return JSONObject(json).toString()
+      return try {
+        JSONObject(json).toString()
+      } catch(e: JSONException) {
```
Comment (BenHenning) [inline data/src/test/java/org/oppia/data/backends/ApiUtils.kt]: Nit: ``catch (``

**Review by BenHenning — APPROVED** [review]
Thanks @vinitamurthi! Let's reference #295 in this PR, as well.

**Review by kevinlee12 — COMMENTED** [review]
config looks good!

**Pull request merged**